juniper/src/ast.rs

```diff
@@ -13,19 +13,19 @@ use crate::{
 /// This enum carries no semantic information and might refer to types that do
 /// not exist.
 #[derive(Clone, Eq, PartialEq, Debug)]
-pub enum Type<'a> {
+pub enum Type<N> {
```

Comment:
I would change this type in a follow-up PR. The type has a recursive definition but in fact GraphQL types are not tree-like in nature:

If I'm not mistaken these are the only possibilities:

* `T`
* `T!`
* `[T]`
* `[T!]`
* `[T]!`
* `[T!]!`

The `Box<Type<N>>` here makes this type awkward to work with, when you want a borrowed "view" of the owned strings. See code comment for `model::DynType`.